apps/oracles/gecko-terminal/src/lib.rs

```diff
@@ -94,16 +148,29 @@ fn get_resources_from_settings(settings: &Settings) -> Result<Vec<FeedConfig>> {
     for feed_setting in &settings.data_feeds {
         //TODO: (EmilIvanichkovv) This is temporary solution
         if feed_setting.data.contains("pool") {
-            let feed_config = serde_json::from_str::<FeedArguments>(&feed_setting.data)
-                .context("Couldn't parse data feed")?;
-            config.push(FeedConfig {
-                id: feed_setting.id.clone(),
-                arguments: GeckoTerminalPool {
-                    network: feed_config.arguments.network,
-                    pool: feed_config.arguments.pool,
-                    reverse: feed_config.arguments.reverse,
-                },
-            })
+            let feed_config_result = serde_json::from_str::<FeedArguments>(&feed_setting.data);
+            match feed_config_result {
+                Ok(feed_config) => {
+                    let c = FeedConfig {
+                        id: feed_setting.id.clone(),
+                        arguments: vec![GeckoTerminalPool {
+                            network: feed_config.arguments.network,
+                            pool: feed_config.arguments.pool,
+                            reverse: feed_config.arguments.reverse,
+                        }],
+                    };
+                    config.push(c);
+                }
+                Err(_) => {
+                    let feed_config = serde_json::from_str::<FeedArgumentsVec>(&feed_setting.data)
+                        .context("Couldn't parse data feed")?;
+                    let c = FeedConfig {
+                        id: feed_setting.id.clone(),
+                        arguments: feed_config.arguments,
+                    };
+                    config.push(c);
+                }
+            };
```

Comment:
Perhaps we should decide that the arguments for this oracle will always be a Vec. This code snippet seems like an overcomplication, and we can simply update the configurations instead.

This can be addressed with the following PR.